gymnasium/envs/mujoco/ant_v5.py

```diff
@@ -0,0 +1,427 @@
+        assert terminated == (
+            not self.is_healthy if self._terminate_when_unhealthy else False
+        )
+        return terminated
+
+    def step(self, action):
+        xy_position_before = self.data.body(self._main_body).xpos[:2].copy()
+        # TODO remove after validation
+        # assert (xy_position_before == self.get_body_com("torso")[:2].copy()).all()
+        self.do_simulation(action, self.frame_skip)
+        xy_position_after = self.data.body(self._main_body).xpos[:2].copy()
+        # TODO remove after validation
+        # assert (xy_position_after == self.get_body_com("torso")[:2].copy()).all()
+
+        xy_velocity = (xy_position_after - xy_position_before) / self.dt
+        x_velocity, y_velocity = xy_velocity
+
+        forward_reward = x_velocity * self._forward_reward_weight
+        healthy_reward = self.healthy_reward
+
+        rewards = forward_reward + healthy_reward
+
+        ctrl_cost = self.control_cost(action)
+        contact_cost = self.contact_cost
+        costs = ctrl_cost + contact_cost
+
+        terminated = self.terminated
+        observation = self._get_obs()
+        info = {
+            "reward_forward": forward_reward,
+            "reward_ctrl": -ctrl_cost,
+            "reward_contact": -contact_cost,
+            "reward_survive": healthy_reward,
+            "x_position": self.data.qpos[0],
+            "y_position": self.data.qpos[1],
+            "distance_from_origin": np.linalg.norm(self.data.qpos[0:2], ord=2),
+            "x_velocity": x_velocity,
+            "y_velocity": y_velocity,
+        }
+
+        reward = rewards - costs
+
+        if self.render_mode == "human":
+            self.render()
+        return observation, reward, terminated, False, info
+
+    def _get_obs(self):
+        position = self.data.qpos.flat.copy()
+        velocity = self.data.qvel.flat.copy()
+
+        if self._exclude_current_positions_from_observation:
+            position = position[2:]
+
+        if self._include_cfrc_ext_in_observation:
+            assert (
+                self.contact_forces[0].flat.copy() == 0
+            ).all()  # TODO remove after validation
+            contact_force = self.contact_forces[1:].flat.copy()
+            assert len(contact_force) == 78
+            return np.concatenate((position, velocity, contact_force))
+        else:
+            return np.concatenate((position, velocity))
+
+    def reset_model(self):
+        noise_low = -self._reset_noise_scale
+        noise_high = self._reset_noise_scale
+
+        qpos = self.init_qpos + self.np_random.uniform(
+            low=noise_low, high=noise_high, size=self.model.nq
+        )
+        qvel = (
+            self.init_qvel
+            + self._reset_noise_scale * self.np_random.standard_normal(self.model.nv)
+        )
+        self.set_state(qpos, qvel)
+
+        observation = self._get_obs()
+
+        return observation
+
+    def _get_reset_info(self):
+        return {
```

Comment:
We might want to include all of the reward based info with zeros. This is particularly helpful for vector environments

Reply:
In general with RL algorithms a reward is not generated during reset.

Also putting in reset `info["reward_survive"] = 0` would be inaccurate.

I am not sure i understand why that would be useful